back/app/services/admin_publications_filtering_service.rb

```diff
@@ -20,23 +20,23 @@ def for_homepage_filter(scope)
     scope.where(publication_type: Project.name)
   end
 
-  # This filter removes AdminPublications that represent folders which contain *only* projects which should not be visible to the current user.
-  # Here we are concerned with 'visibility' in terms of the Project.visible_to attribute, which can have one of 3 values: public, groups or admins.
-  add_filter('remove_not_allowed_parents') do |visible_publications, options|
-    next visible_publications unless ['true', true, '1'].include? options[:remove_not_allowed_parents]
-
-    public_project_ids          = Project.publicly_visible.ids
-    public_project_publications = AdminPublication.includes(:parent).where(publication_id: public_project_ids).where.not(parent_id: nil)
-
-    visible_or_public_publication_ids = visible_publications.map(&:parent_id).concat(public_project_publications.map(&:parent_id)).compact.uniq
-
-    parents_with_visible_children = visible_publications.where(id: visible_or_public_publication_ids)
-    parents_without_any_children  = visible_publications.where(children_allowed: true, children_count: 0)
-    non_parents                   = visible_publications.where(children_allowed: false)
-
-    parents_with_visible_children.or(parents_without_any_children)
+  # This filter removes AdminPublications that represent folders,
+  # which contain *only* projects which should not be visible to the current user.
+  add_filter('remove_not_allowed_parents') do |scope, options|
+    next scope unless ['true', true, '1'].include? options[:remove_not_allowed_parents]
+
+    parents_with_visible_children_publication_ids = AdminPublication.includes(:parent)
+      .where(publication_id: Project.publicly_visible.not_draft.ids.uniq)
+      .where.not(parent_id: nil)
+      .map(&:parent_id)
+
+    parents_with_visible_children = scope.where(id: parents_with_visible_children_publication_ids)
+    parents_without_any_children  = scope.where(children_allowed: true, children_count: 0)
+    non_parents                   = scope.where(children_allowed: false)
+
+    parents_with_visible_children
+      .or(parents_without_any_children)
       .or(non_parents)
-      .or(public_project_publications)
```

Comment:
Why did we have it before and what has changed? :smile:

Reply:
These are the visible projects inside folders (visibility "public" and non-draft). They should be covered with `non_parents`, right?

Reply:
@alexander-cit correct?

Reply:
Yes, I guess you're right. I'm just not sure why we've added it before. But probably it was just a mistake.